Sources/SwiftDocC/Utility/FoundationExtensions/String+Capitalization.swift

```diff
@@ -0,0 +1,30 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+extension StringProtocol {
+    
+    /// Returns the string with the first letter capitalized.
+    /// This auto-capitalization only occurs if the first word is all lowercase and contains only characters A-Z.
+    /// The first word can also contain punctuation (e.g. a period, comma, hyphen, semi-colon, colon).
+    var capitalizeFirstWord: String {
+        let firstWord = self.components(separatedBy: " ").first ?? ""
```

Comment:
I think using [`CharacterSet.whitespacesAndNewlines`](https://developer.apple.com/documentation/foundation/characterset/1779801-whitespacesandnewlines ) would be more robust.

Reply:
Also if there are no spaces wouldn't the first word be the entire string?

Reply:
You could also avoid some temporary allocations by using substring APIs. Something like:
```swift
let firstWord = prefix(while: { !$0.isWhitespace })
guard let firstLetter = firstWord.first,
      firstLetter.isLowercase,
      !firstWord.contains(where: \.isUppercase)
else {
    return String(self)
}
```

Reply:
Adjusted this to use a combination of this and Daniel's suggestion below to use CharacterSet APIs! Let me know what you think about the new implementation @daniel-grumberg @d-ronnqvist